internal/speedtest/url_download.go

```
@@ -0,0 +1,382 @@
+			case <-progressDone:
+				return
+			case <-ticker.C:
+				mu.Lock()
+				elapsed := time.Since(startTime).Seconds()
+				currentSpeed := 0.0
+				if elapsed > 0 {
+					currentSpeed = float64(totalBytes) * 8 / elapsed / 1000000 // Mbps
+				}
+
+				// Calculate real progress
+				progress := 50.0 // Default fallback
```

golangci-lint (2.12.2)

[error] ineffectual assignment to progress

(ineffassign)